app/controllers/schedules_controller.rb

```diff
@@ -49,6 +49,7 @@ def new
           'You must save the Schedule first before each Task is assigned an ID.'
     flash.now[:info] = msg
     @schedule = params[:schedule].nil? ? Schedule.new : Schedule.new(schedule_params)
+    @root = Task.new
```

Comment:
If you're not using `@root` in the view, I think you can move the creation of this root task into the Schedule #initialize method.

Doing that would also replace Schedule#assign_root and (hopefully) make it so you don't have to save twice on create.